rmf_traffic/test/unit/test_BroadPhase.cpp

```diff
@@ -0,0 +1,302 @@
+      std::cout << "Two non-overlapping L-shaped trajectories\n";
+      std::cout << "Total: " << sec << std::endl;
+      std::cout << "Per run: " << sec/N << std::endl;
+    }
+  }
+
+  GIVEN("Two overlapping parallel straight-line trajectories with n segments")
+  {
+    int n_seg = 200;
+
+    rmf_traffic::Trajectory trajectory_a("test_map");
+    rmf_traffic::Trajectory trajectory_b("test_map");
+
+    for (auto i = 0; i < n_seg; i++)
+    {
+      trajectory_a.insert(
+      begin_time + std::chrono::seconds(i * 5),
+      profile,
+      Eigen::Vector3d{0.0 + i*10.0, 10.0, 0},
+      Eigen::Vector3d{0.0, 0.0, 0.0});
+
+      trajectory_b.insert(
+      begin_time + std::chrono::seconds(i * 5),
+      profile,
+      Eigen::Vector3d{0.0 + i*10.0, 9.0, 0},
+      Eigen::Vector3d{0.0, 0.0, 0.0});
+    }
+    REQUIRE(trajectory_a.size() == n_seg);
+    REQUIRE(trajectory_b.size() == n_seg);
+
+    const auto start_time = std::chrono::steady_clock::now();
+    for (std::size_t i = 0; i < N; ++i)
+    {
+      CHECK(rmf_traffic::DetectConflict::broad_phase(
+          trajectory_a,
+          trajectory_b));    
+    }
+    const auto end_time = std::chrono::steady_clock::now();
+
+    if (test_performance)
+    {
+      const double sec = rmf_traffic::time::to_seconds(end_time - start_time);
+      std::cout << "Straight-line trajectories with " << n_seg << " segments\n";
+      std::cout << "Total: " << sec << std::endl;
+      std::cout << "Per run: " << sec/N << std::endl;
+    }
+  }
+
+  GIVEN("Two non-overlapping parallel straight-line trajectories with n segments")
+  {
+    int n_seg = 200;
+
+    rmf_traffic::Trajectory trajectory_a("test_map");
+    rmf_traffic::Trajectory trajectory_b("test_map");
+
+    for (auto i = 0; i < n_seg; i++)
+    {
+      trajectory_a.insert(
+      begin_time + std::chrono::seconds(i * 5),
+      profile,
+      Eigen::Vector3d{0.0 + i*10.0, 10.0, 0},
+      Eigen::Vector3d{0.0, 0.0, 0.0});
+
+      trajectory_b.insert(
+      begin_time + std::chrono::seconds(i * 5),
+      profile,
+      Eigen::Vector3d{0.0 + i*10.0, -10.0, 0},
+      Eigen::Vector3d{0.0, 0.0, 0.0});
+    }
+    REQUIRE(trajectory_a.size() == n_seg);
+    REQUIRE(trajectory_b.size() == n_seg);
+
+    const auto start_time = std::chrono::steady_clock::now();
+    for (std::size_t i = 0; i < N; ++i)
+    {
+      CHECK_FALSE(rmf_traffic::DetectConflict::broad_phase(
+          trajectory_a,
+          trajectory_b));    
+    }
+    const auto end_time = std::chrono::steady_clock::now();
+
+    if (test_performance)
+    {
+      const double sec = rmf_traffic::time::to_seconds(end_time - start_time);
+      std::cout << "Non-overlapping straight-line trajectories with " << n_seg << " segments\n";
+      std::cout << "Total: " << sec << std::endl;
+      std::cout << "Per run: " << sec/N << std::endl;
+    }
+  }
+
+  GIVEN("Time ranges and bounding boxes overlap but segments do not intersect")
```

Comment:
What about this same test but the time of `trajectory_b` is offset just enough that the bounding boxes don't intersect when time is accounted for?

Reply:
The test label is not clear but the `time ranges` and `bounding boxes` referred to are of the entire trajectories. The segments are already defined such that their bounding boxes do not intersect when time is accounted for. I could make the time differences more apparent perhaps?

Reply:
Oh okay! Yep, after looking at the test more carefully, it does seem to be doing what I was suggesting, thanks!

Maybe we could rephrase the test description to something like "Bounding boxes overlap in space, but are offset in time"?

Reply:
That's a much better label, will update accordingly. Thanks!